vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringHelper.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2000-2017 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.spring;
+
+import java.util.Iterator;
+import java.util.Locale;
+import java.util.Objects;
+
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * Utility methods for spring
+ */
+public class SpringHelper {
+
+    private SpringHelper() {
+    }
+
+    /**
+     * Gets the property name used after being parsed by Spring Boot, accepting
+     * relaxed parsing
+     *
+     * @param env the environment, not {@code null}
+     * @param propertyName the relaxed property name, not {@code null}
+     * @return the stored property name. {@code null} if not found
+     */
+    public static String getStoredPropertyName(Environment env,
```

Comment:
If method may return `null` then it's better to return `Optional`: it makes the contract clearer.